config/environments/production.rb

```diff
@@ -125,18 +126,24 @@
 
   # Adds plugin assets to the application digests hash if a manifest file exists for a plugin
   config.after_initialize do
-    ::Rails::Engine.subclasses.map(&:instance).each do |engine|
-      [engine.root, app.root].each do |root_dir|
-        manifest_path = File.join(root_dir, "public/assets/#{engine.engine_name}/manifest.yml")
-
-        if File.file?(manifest_path)
-          assets = YAML.load_file(manifest_path)
-
-          assets.each_pair do |file, digest|
-            config.assets.digests[file] = digest
+    if manifest_file = Dir.glob('public/assets/manifest*.json').first
+      foreman_manifest = JSON.parse(File.read(manifest_file))
+      ::Rails::Engine.subclasses.map(&:instance).each do |engine|
+        [engine.root, app.root].each do |root_dir|
+          manifest_path = File.join(root_dir, "public/assets/#{engine.engine_name}/#{engine.engine_name}.json")
+          if File.file?(manifest_path)
+            assets =  JSON.parse(File.read(manifest_path))
+            foreman_manifest['files'] = foreman_manifest['files'].merge(assets['files'])
+            foreman_manifest['assets'] = foreman_manifest['assets'].merge(assets['assets'])
           end
         end
       end
+
+      manifest_file = Tempfile.new(['complete_manifest', '.json'])
```

Comment:
Is this going to generate a new temporary file every time the app starts?  (Including in workers under Passenger etc.)  Should it be shared in tmp/, with an exclusive open to write it?  Or cleaned up somehow.